core/src/main/java/org/apache/iceberg/stats/StatsCompressionCodec.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import java.util.Objects;
+
+public enum StatsCompressionCodec {
+  /**
+   * LZ4 single compression frame with content size present
+   */
+  LZ4("lz4"),
+
+  /**
+   * Zstandard single compression frame with content size present
+   */
+  ZSTD("zstd"),
+  /**/;
+
+  private final String codecName;
+
+  StatsCompressionCodec(String codecName) {
+    this.codecName = codecName;
+  }
+
+  public String getCodecName() {
+    return codecName;
+  }
+
+  public static StatsCompressionCodec forName(String codecName) {
+    Objects.requireNonNull(codecName, "codecName is null");
```

Comment:
Nit: Usually we use the relocated `Preconditions` form.

Also, does it make more sense to just maintain a mapping between the two vs iterating over them?

Reply:
from perf perspective, for enum with two elements, i don't know which is actually faster
from readability perspective, it's probably subjective which is more readable, so which one you'd prefer?

Reply:
For 2 elements actually it’s probably overkill in my opinion. Good point.